cedar-policy-core/src/ast/policy.rs

```diff
@@ -1163,16 +1167,51 @@ impl From<BTreeMap<AnyId, Annotation>> for Annotations {
 /// Struct which holds the value of a particular annotation
 #[derive(Serialize, Deserialize, Clone, Hash, Eq, PartialEq, Debug, PartialOrd, Ord)]
 pub struct Annotation {
-    /// Annotation value
-    pub val: SmolStr,
+    /// Annotation value. `None` for annotations without a value, i.e., `@foo`.
+    /// An annotation without a value should be treated as equivalent to the
+    /// value being `""`. This interpretation is implemented by the `AsRef<str>`
+    /// impl an `val` method below.
+    raw_val: Option<SmolStr>,
```

Comment:
Why `Option` and explicitly saying that `None` and `Some("")` are equivalent?  This seems like a footgun for later.  Wouldn't it be better to just have a `SmolStr` here so that there's only one representation?

Reply:
The reason is that conversion to EST, which is currently written to go through this AST struct, even when converting directly from CST, should be lossless, so this struct needed to know if it was originally present.

Reply:
Personally I would propose that we pick one or the other representation to be canonical and output that EST.  So either we decide that empty annotation values are always omitted, or always explicit `""`.  I believe "lossless" already drops extraneous parens, is this any different?

Reply:
A related decision: In the EST I updated annotations so that `null` is an allowed value, equivalent to `""`. Thoughts? I think we would eventually need to allow that in a future world where we differentiate no-value and empty-string, but it's not necessary for now.

Reply:
I could see an argument either way there.  Some sugar in the EST can be helpful, but the feature request was probably primarily for sugar in the Cedar policy format, and it's totally reasonable to require explicitness in the EST.

Reply:
Made some of the cst-to-ast code generic so that est-to-ast can still reuse the same id and string validation checks without passing through the ast structs to avoid this issue